bundles/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/internal/handler/NukiBridgeHandler.java

```diff
@@ -40,55 +52,111 @@
  * sent to one of the channels.
  *
  * @author Markus Katter - Initial contribution
+ * @contributer Jan Vybíral - Improved callback handling
  */
+@NonNullByDefault
 public class NukiBridgeHandler extends BaseBridgeHandler {
 
     private final Logger logger = LoggerFactory.getLogger(NukiBridgeHandler.class);
     private static final int JOB_INTERVAL = 600;
 
-    private HttpClient httpClient;
+    private final HttpClient httpClient;
+    @Nullable
     private NukiHttpClient nukiHttpClient;
-    private String callbackUrl;
+    @Nullable
+    private final String callbackUrl;
+    @Nullable
     private ScheduledFuture<?> checkBridgeOnlineJob;
+    @Nullable
     private String bridgeIp;
     private boolean manageCallbacks;
     private boolean initializable;
 
-    public NukiBridgeHandler(Bridge bridge, HttpClient httpClient, String callbackUrl) {
+    public NukiBridgeHandler(Bridge bridge, HttpClient httpClient, @Nullable String callbackUrl) {
         super(bridge);
         logger.debug("Instantiating NukiBridgeHandler({}, {}, {})", bridge, httpClient, callbackUrl);
-        this.httpClient = httpClient;
         this.callbackUrl = callbackUrl;
+        this.httpClient = httpClient;
+
+        // initialize config from discovered propeties
+        initConfigFromProperty(NukiBindingConstants.PROPERTY_BRIDGE_IP, NukiBindingConstants.CONFIG_IP,
+                String::toString);
```

Comment:
What's the reason for retrieving the config from properties? Does it work with manually added Things, too?

By utilizing `getConfigAs()`, the framework takes care about the correct types.